### PR TITLE
Ensure AppLocalResolver handles package with multiple libs

### PR DESCRIPTION
**Pull request by ericstj**
This was failing to pass references to a package which contains more than one lib.

I debugged this failure in the 5.0-release branch and root-caused to this, which should be fixed regardless.  If you want you can port this to 5.0 as well.

Essentially a port of https://github.com/dotnet/extensions/commit/7aff1be4fc62f6c5534801b40272319334e0636a to a copy of the same type.  cc @pakrym @dougbu @eerhardt (as DependencyContext expert 😉)

### CODE REVIEW

**Review by dougbu — APPROVED** [review]
Thanks very much for getting to this fix before I did❕ Think a `cherry-pick` back to release/5.0 will definitely be in order once this is in.

**Comment by ericstj** [comment]
One thing you could even consider after this (probably not for release) would be switching these tests to use https://github.com/dotnet/runtime/blob/035ebeab160b9a281e5fbb3fa0c237f244c81b7a/src/libraries/Microsoft.Extensions.DependencyModel/src/Resolution/AppBaseCompilationAssemblyResolver.cs.  It's not clear to me why they aren't using this as it provides similar functionality.

**Comment by dougbu** [comment]
/fyi @ajaybhargavb @pranavkm @dotnet/aspnet-build this should avoid test failures when Microsoft.Extensions.Internal.Transport ref/ assemblies appear (but aren't really) newer than the implementation assemblies in individual Extensions packages.

**Review by eerhardt — APPROVED** [review]
This looks good. I've fixed similar bugs in DependencyModel in the past.

**Comment by dougbu** [comment]
> It's not clear to me why they aren't using this as it provides similar functionality.

I don't have the whole history but suspect either awareness or timing e.g. `DiagnosticProject` may have been around longer than `AppBaseCompilationAssemblyResolver`.

**Comment by pranavkm** [comment]
> It's not clear to me why they aren't using this as it provides similar functionality.

Very likely we were unaware of this.

**Review by pranavkm — APPROVED** [review]

**Comment by ericstj** [comment]
I'll defer to you guys to merge this one.

**Comment by dougbu** [comment]
Switched to release/5.0 and brought in a fix from master allowing local testing in this area (and others)

File: src/Analyzers/Microsoft.AspNetCore.Analyzer.Testing/src/DiagnosticProject.cs [hunk 1]
```diff
@@ -25,6 +25,7 @@ public class DiagnosticProject
         /// </summary>
         public static string TestProjectName = "TestProject";
 
+        private static readonly ICompilationAssemblyResolver _assemblyResolver = new AppBaseCompilationAssemblyResolver();
```
Comment (dougbu) [inline src/Analyzers/Microsoft.AspNetCore.Analyzer.Testing/src/DiagnosticProject.cs:28]: This uses the app base directory by default. That works locally and should work everywhere&hellip;

**Comment by dougbu** [comment]
@Pilchie would you prefer I back-port 3a806b04362a separately❔

@dotnet/aspnet-build please add your reviews since this goes beyond what @pranavkm, @eerhardt, and I signed off on.

**Comment by dougbu** [comment]
@ericstj `AppBaseCompilationAssemblyResolver` was a great suggestion❕ And, thanks for your branch😺

**Comment by Pilchie** [comment]
No, I'm okay with bringing all of this.

**Review by wtgodbe — APPROVED** [review]
Looks good to me

**Comment by dougbu** [comment]
Again, much appreciated @ericstj

**Comment by ericstj** [comment]
Glad to help :)

**Pull request merged**